### PR TITLE
Usage of pod and unix-sockets

### PR DESCRIPTION
**Pull request by luebken**
This PR shows how to use a pod in order to use unix-sockets as the communication between the app and the redis.

Orga question: Should the pod example stay in this branch or move to its own repo?

TODOs:
 * [ ] Finish Readme
 * [ ] Decide on repo naming

### CODE REVIEW

**Comment by luebken** [comment]
RFC @marians @denderello

File: main.go [hunk 1]
```diff
@@ -34,9 +33,10 @@ type WeatherReport struct {
 func main() {
 	var err error
 	log.Println("Establishing connection to Redis")
-	redisCon, err = redis.Dial("tcp", redisAddress())
+
```
Comment (luebken) [inline main.go]: TODO cleanup

File: main.go [hunk 2]
```diff
@@ -34,9 +33,10 @@ type WeatherReport struct {
...
+	redisCon, err = redis.Dial("unix", "/tmp/redis.sock")
 	if err != nil {
-		log.Fatalf("Could not connect to Redis with error: %s", err)
+		log.Fatalf("Could not connect to Redis with error: %s\n", err)
```
Comment (luebken) [inline main.go]: TODO remove newline

**Comment by marians** [comment]
From your question I guess that you don't intent to overwrite the current master, right? Good.

Leaving it in a branch in this repository makes this virtually invisible. That would be sad. So I definitely vote for taking it into it's own repository.

**Comment by luebken** [comment]
More repos also means more maintaince. But I would also go for an own repo currently to market our pod capabilities. 




What about naming? giantswarm-firstpod-go?



-- 

Giant Swarm GmbH · c/o Startplatz · Mediapark 5 · D-50670 Köln

Mobile: +49 176 21236361 · @luebken · giantswarm.io

Handelsregister: Amtsgericht Köln, HRB 81484· Sitz der Gesellschaft: Köln

Geschäftsführer: Timo Derstappen, Henning Lange, Oliver Thylmann

**Comment by marians** [comment]
I think it's mostly the amount of code that influences the amount of maintenance work, less the number of repositories.

How about plain simple `giantswarm-pod-example` or `giantswarm-pod-examples` (in case we might add other languages/examples later)?

**Comment by luebken** [comment]
The idea was that we might want to port this example `giantswarm-firstpod-go` and `giantswarm-firstpod-nodejs` into different repos.

**Comment by marians** [comment]
I can't imagine that the language differences are of any importance. Probably most users will find out how to write to or read from a socket in their language of choice.

I'd prefer to put that effort into creating additional meaningful examples, in what language ever.

**Comment by luebken** [comment]
RFR @ewoutp

**Comment by luebken** [comment]
I would like to get a technical LGTM first and merge it into the `pod` branch.

File: Makefile [hunk 3]
```diff
@@ -36,22 +36,24 @@ currentweather: $(GO_SOURCE)
 # Building your custom docker image
 docker-build: currentweather
 	docker build -t registry.giantswarm.io/$(GIANTSWARM_USERNAME)/currentweather-go .
+	docker build -t registry.giantswarm.io/$(GIANTSWARM_USERNAME)/redis -f Dockerfile-redis .
```
Comment (ewoutp) [inline Makefile]: I would vote to make the redis name a bit more specific, especially since it is a customized image.
This will cause conflicts.

**Comment by ewoutp** [comment]
The code looks ok to me.
One thing that is a bit strange to me it the use of the `/tmp` directory. Using it for sockets (like you do) is fine, but since we mount a volume to it, I would call it differently. Mounting a volume to `/tmp` seems wierd to me.

And why not pass the socket name through a command line flag? A tiny bit more work in main.go, but a lot nicer since everything then comes together in swarm.json.

**Comment by ewoutp** [comment]
Finally a though on branch/repos.

Why not use the same repo, but put this pod example in a subdirectory (e.g. `/using-unix-sockets`)

File: Makefile (hunk 3 above)
Comment (luebken) [inline Makefile]: Agree.

Ideas `redis-unixsocket`, `redis-uds`. I tend to go with `redis-uds`.

**Comment by luebken** [comment]
> I would call it differently. Mounting a volume to /tmp seems wierd to me.

Good point. The internets aren't sure. Actually `tmp`was a valid choice. But I will try `/var/run/`
http://unix.stackexchange.com/questions/88083/idiomatic-location-for-file-based-sockets-on-debian-systems
http://stackoverflow.com/questions/7580346/where-to-place-unix-domain-af-unix-sockets-end-points-files

**Comment by luebken** [comment]
>And why not pass the socket name through a command line flag?
+1 Will do.

File: Makefile (hunk 3 above)
Comment (ewoutp) [inline Makefile]: I would be specific, so `redis-unixsocket` seems fine to me

**Comment by luebken** [comment]
Closing this in favour of https://github.com/giantswarm/giantswarm-pod-example/pull/1

**Pull request closed without merging**